### PR TITLE
machines: Enable interface type option 'bridge' for NICs

### PR DESCRIPTION
**Pull request by skobyda**
Screenshot from 2019-09-12 13-29-51

### CODE REVIEW

**Comment by martinpitt** [comment]
@KKoukiou: We need to weigh "installation of cockpit-machines pulls in NetworkManager" (which may be undesirable on a lot of servers, as they often use something more static like /etc/sysconfig/network/, ifupdown, or systemd-networkd) against not discovering that feature. My gut feeling is that a `Suggests:` would be better, if the UI can give some hint that these interface types are only available with NetworkManager. But the UI needs that anyway IMHO -- even with Recommends it's still not a given that NM is installed, or enabled, or used to manage that particular bridge.

What exactly do you need here? You don't need NM for merely enumerating network devices and getting notified about changes -- that can be done in /sys/class/net/ and uevents. And I suppose the Machines page shouldn't *change* these interfaces on the host?

**Comment by skobyda** [comment]
@martinpitt Yes the point is to list all the interfaces, and perhaps in some cases filter only devices which are bridges.
I don't think we need to be notified about changes, as we only show network devices in dropdown list in some modals. At this places, network devices can be enumerated on the spot. So we don't care about any changes made to network devices.

**Comment by skobyda** [comment]
So I replaced how we get list of network bridges. Instead of calling libvirt API, I call $brtcl show and parse the output.
PR is ready for another review.

**Comment by KKoukiou** [comment]
@skobyda not sure if you have noticed - the tests are breaking with this one.
See https://209.132.184.41:8493/logs/pull-12780-20190925-164040-32c7c2bb-cockpit-project-cockpit--debian-stable/log.html#237-2

File: pkg/machines/components/vmnetworktab.jsx [hunk 1]
```diff
@@ -37,11 +38,32 @@ class VmNetworkTab extends React.Component {
...
+    open() {
+        this.setState({ showModal: true });
     }
 
     componentDidMount() {
+        cockpit.spawn(["ls", "/sys/class/net"])
+                .fail(e => console.log(e))
+                .done(output => {
+                    const devs = output.split('\n');
+                    devs.pop();
```
Comment (KKoukiou) [inline pkg/machines/components/vmnetworktab.jsx]: Can you leave me a comment why we need the pop here :) I don't really see through it.

**Comment by KKoukiou** [comment]
For release notes: (pls correct wording if not appropriate)
`Allow setting up Network Interfaces which provide a bridge from the VM directly to the LAN. This provides the guest VM full incoming & outgoing net access just like a physical machine.`

Screen Shot 2019-12-13 at 15 54 35
Screen Shot 2019-12-13 at 15 54 16

**Pull request merged**